### PR TITLE
Fix project build

### PR DESCRIPTION
**Pull request by pablochacin**
Fix how the project is build adopting the same approach than  dex-operator:
- Makefile
- circleCi configuration
- Ancillary files

### CODE REVIEW

**Comment by pablochacin** [comment]
@MalloZup I already rebased to the master this morning.

File: pkg/controller/registry/registry_controller_suite_test.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ var cfg *rest.Config
 
 func TestMain(m *testing.M) {
 
+
 	var t *envtest.Environment
```
Comment (MalloZup) [inline pkg/controller/registry/registry_controller_suite_test.go]: why do we add a newline?

File: .circleci/config.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ jobs:
 
       - run:
           name: Build the Docker image
-          command: make clean docker-image
+          command: make clean _vendor-download docker-image
```
Comment (MalloZup) [inline .circleci/config.yml]: do we have this target?

Comment (pablochacin) [reply .circleci/config.yml]: It was missing, I included it in the Makefile

File: pkg/apis/apis.go [hunk 3]
```diff
@@ -15,6 +15,9 @@
  *
  */
 
+// Generate deepcopy for apis (and do not use modules: it makes it slow)
+//go:generate sh -c "GO111MODULE=on deepcopy-gen -O zz_generated.deepcopy -i ./... -h ../../../../hack/boilerplate.go.txt"
```
Comment (ereslibre) [inline pkg/apis/apis.go]: The route to `boilerplate.go.txt` also needs updating: `../../hack/boilerplate.go.txt` :)

**Pull request merged**